src/network/wire/lib.zig

```diff
@@ -308,6 +311,66 @@ test "ok_send_ping_message" {
     }
 }
 
+test "ok_send_merkleblock_message" {
+    const Config = @import("../../config/config.zig").Config;
+    const ArrayList = std.ArrayList;
+    const test_allocator = std.testing.allocator;
+    const MerkleBlockMessage = protocol.messages.MerkleBlockMessage;
+    const network_id = Config.BitcoinNetworkId.MAINNET;
+
+    var list: std.ArrayListAligned(u8, null) = ArrayList(u8).init(test_allocator);
+    defer list.deinit();
+
+    const block_header = BlockHeader{
+        .version = 1,
+        .prev_block = [_]u8{0} ** 32,
+        .merkle_root = [_]u8{1} ** 32,
+        .timestamp = 1234567890,
+        .bits = 0x1d00ffff,
+        .nonce = 987654321,
+    };
+    const hashes = try test_allocator.alloc([32]u8, 3);
+
+    const flags = try test_allocator.alloc(u8, 1);
+    const transaction_count = 1;
+    const message = MerkleBlockMessage.new(block_header, transaction_count, hashes, flags);
+
+    defer test_allocator.free(hashes);
+    defer test_allocator.free(flags);
+    // Fill in the header_hashes
+    for (message.hashes) |*hash| {
+        for (hash) |*byte| {
+            byte.* = 0xab;
+        }
+    }
+    for (flags) |*f| {
+        f.* = 0x1;
+    }
+    const serialized = try message.serialize(test_allocator);
+    defer test_allocator.free(serialized);
+
+    const deserialized = try MerkleBlockMessage.deserializeSlice(test_allocator, serialized);
+    defer deserialized.deinit(test_allocator);
+
+    const writer = list.writer();
+    try sendMessage(test_allocator, writer, Config.PROTOCOL_VERSION, Config.BitcoinNetworkId.MAINNET, message);
+    var fbs: std.io.FixedBufferStream([]u8) = std.io.fixedBufferStream(list.items);
+    const reader = fbs.reader();
+
+    const received_message = try receiveMessage(test_allocator, reader, network_id) orelse unreachable;
```

Comment:
there is a `write_and_read_message` in the file that does exactly that

Reply:
thank you very much, very appreciated your reviews.